datachannel.go

```diff
@@ -286,6 +287,30 @@ func (d *DataChannel) handleOpen(dc *datachannel.DataChannel) {
 	}
 }
 
+// OnError sets an event handler which is invoked when
+// the underlying data transport cannot be read.
+func (d *DataChannel) OnError(f func()) {
+	d.mu.Lock()
+	defer d.mu.Unlock()
+	d.onErrorHandler = f
+}
+
+func (d *DataChannel) onError() (done chan struct{}) {
+	d.mu.RLock()
+	hdlr := d.onErrorHandler
+	d.mu.RUnlock()
+
+	done = make(chan struct{})
+	if hdlr == nil {
+		close(done)
+		return
+	}
+
+	hdlr()
```

Comment:
We want to run all the callbacks in a goroutine, otherwise a user may block the library internals 

```
    go func() {
        hdlr(err)
        close(done)
    }()
```